core/platform/search/elastic_search_services.py

```diff
@@ -107,47 +106,54 @@ def clear_index(index_name):
 
 
 def search(
-        query_string, index_name, offset=None,
-        limit=feconf.SEARCH_RESULTS_PAGE_SIZE, ids_only=False):
+        query_string, index_name, cursor=0,
```

Comment:
Why not call it "offset" instead of "cursor"? A cursor is typically some kind of encoded string.

(Note: it's still important to ensure this is consistent between the implementations.)

Reply:
I wanted to keep the original implementation since currently cursor works for the old case but since the we're going to use offset from now on, ill change it to offset, this makes it slightly confusing for testing since currently all use cases use offset => Ill change the method signature to offset but keep the input a cursor for now until we can change it bc it doesn't make sense to name a cursor string an offset.

Reply:
Ah, I missed this before writing some of the other comments, sorry. We may have to resolve this one first before resolving the others.

Here is how I would suggest going about this:
- In the common API, add args for offset and cursor. These should both default to None.
- The ES impl of search() should assert that the arg for cursor is None before doing anything else.
- The GAE impl of search() should assert that the arg for offset is None before doing anything else.
- Add a TODO in the docstring for the platform method, as well as methods/functions in the domain layer above it which make use of the cursor, to either (a) delete the cursor arg completely or (b) switch over to using offset, once the switchover to ES in the platform layer is done. The choice of whether to specify (a) or (b) depends on the context. Let's maybe have a single issue for this -- you could put the instructions you've written above there, plus a note to address these cases, and assign it to Shubha.
- Finally, in this PR, never compare a cursor and an offset, or assign one to the other's field. This is a recipe for confusion.

Does that sound ok? Let me know if you have any questions about this.

Thanks!

Reply:
Done in https://github.com/oppia/oppia/pull/11295/commits/93a123cc7869c29b5a8b9e86d62b7fecb13f7a5a.